src/Sylius/Component/Payment/spec/Resolver/CompositeMethodsResolverSpec.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\Payment\Resolver;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Component\Payment\Model\PaymentInterface;
+use Sylius\Component\Payment\Model\PaymentMethodInterface;
+use Sylius\Component\Payment\Resolver\MethodsResolverInterface;
+use Sylius\Component\Registry\PrioritizedServiceRegistryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class CompositeMethodsResolverSpec extends ObjectBehavior
+{
+    function let(PrioritizedServiceRegistryInterface $resolversRegistry)
+    {
+        $this->beConstructedWith($resolversRegistry);
+    }
+    
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Component\Payment\Resolver\CompositeMethodsResolver');
+    }
+
+    function it_implements_Sylius_payment_methods_resolver_interface()
+    {
+        $this->shouldImplement(MethodsResolverInterface::class);
+    }
+
+    function it_uses_registry_to_provide_payment_methods_for_payment_subject(
+        MethodsResolverInterface $firstMethodsResolver,
+        MethodsResolverInterface $secondMethodsResolver,
+        PrioritizedServiceRegistryInterface $resolversRegistry,
+        PaymentMethodInterface $paymentMethod,
+        PaymentInterface $payment
+    ) {
+        $resolversRegistry->all()->willReturn([$firstMethodsResolver, $secondMethodsResolver]);
+
+        $firstMethodsResolver->supports($payment)->willReturn(false);
+        $secondMethodsResolver->supports($payment)->willReturn(true);
+
+        $secondMethodsResolver->getSupportedMethods($payment)->willReturn([$paymentMethod]);
+
+        $this->getSupportedMethods($payment)->shouldReturn([$paymentMethod]);
+    }
+
+    function it_returns_empty_array_if_none_of_registered_resolvers_support_passed_payment_subject(
+        MethodsResolverInterface $firstMethodsResolver,
+        MethodsResolverInterface $secondMethodsResolver,
+        PrioritizedServiceRegistryInterface $resolversRegistry,
+        PaymentInterface $payment
+    ) {
+        $resolversRegistry->all()->willReturn([$firstMethodsResolver, $secondMethodsResolver]);
+
+        $firstMethodsResolver->supports($payment)->willReturn(false);
+        $secondMethodsResolver->supports($payment)->willReturn(false);
+
+        $this->getSupportedMethods($payment)->shouldReturn([]);
+    }
+
+    function it_support_every_payment_subjects(PaymentInterface $subject)
```

Comment:
supports

Also, I think we should remove both Shipping & PaymentSubjectInterfaces. It makes sense for promotions, but here, not that much. Only complicates things, because only reasonable subjects are payment and shipment anyway. Thoughts?